examples/vision/attention_mll_classification.py

```diff
@@ -0,0 +1,579 @@
+
+Some regularization techniques are considered to avoid overfitting the model which ensures
+minimal generalization error.
+"""
+
+
+def train(train_data, train_labels, val_data, val_labels, model):
+
+    # Train model.
+    # Prepare callbacks.
+    # Path where to save best weights.
+    with tempfile.NamedTemporaryFile(suffix=".h5") as tmp:
+
+        # Take the file name from the wrapper.
+        file_path = tmp.name
+
+        # Initialze model checkpoint callback.
+        mc = callbacks.ModelCheckpoint(
+            file_path,
+            monitor="val_loss",
+            verbose=0,
+            mode="min",
+            save_best_only=True,
+            save_weights_only=True,
+        )
+
+        # Initialze early stopping callback.
+        # The model performance is monitored across the unseen data and stops training
+        # when the generalization error cease to decrease
+        es = callbacks.EarlyStopping(monitor="val_loss", patience=10, mode="min")
+
+        # Compile model.
+        model.compile(
+            optimizer="adam",
+            loss="sparse_categorical_crossentropy",
+            metrics=["accuracy"],
+        )
+
+        # Fit model.
+        model.fit(
+            train_data,
+            train_labels,
+            validation_data=(val_data, val_labels),
+            epochs=20,
+            class_weight=calculate_weights(train_labels),
+            batch_size=1,
+            callbacks=[es, mc],
+            verbose=0,
+        )
+
+        # Load best weights.
+        model.load_weights(file_path)
+
+        return model
+
+
+# Building model(s).
+instance_shape = train_data[0][0].shape
+models = [create_model(instance_shape) for _ in range(ENSEMBLE_AVG_COUNT)]
+
+# Training model(s).
+trained_models = [
+    train(train_data, train_labels, val_data, val_labels, model)
+    for model in tqdm(models)
+]
+
+"""
+# Model Evaluation
+
+The models are in their deterministic state now and ready for evaluation.
+
+Based on the number of models (`ENSEMBLE_AVG_COUNT`), the models predict the results
+and then averaged out (equal contribution per model).
+"""
+
+
+def predict(data, labels, trained_models):
+    predictions = []
+    attention_weights = []
+    losses = []
+    accuracies = []
+
+    for model in trained_models:
+
+        # Predict output classes on validation data.
+        pred = model.predict(data)
+        predictions.append(pred)
+
+        # Create intermediate model to get attention layer weights.
+        interm_model = keras.Model(model.input, model.get_layer("alpha").output)
+
+        # Predict attention layer weights.
+        pred_interm = interm_model.predict(data)
+
+        # Reshape list of arrays.
+        a_w = np.squeeze(np.swapaxes(pred_interm, 1, 0))
+        attention_weights.append(a_w)
+
+        loss, acc = model.evaluate(data, labels, verbose=0)
+        losses.append(loss)
+        accuracies.append(acc)
+
+    def average(data_list):
```

Comment:
Just inline this, no need for a separate function

Reply:
Will update it.